common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/TraceTimeIntervalWarningService.java

```diff
@@ -22,46 +29,77 @@
   private static final Logger logger =
       LoggerFactory.getLogger(TraceTimeIntervalWarningService.class);
 
+  /**
+   * Marker constant for the scenario where checkin location id hashing can not be computed.
+   */
+  private static final byte[] NO_HASH = new byte[0];
+
   private final TraceTimeIntervalWarningRepository traceTimeIntervalWarningRepo;
+  private final FakeCheckinsGenerator fakeCheckinsGenerator;
 
   public TraceTimeIntervalWarningService(
-      TraceTimeIntervalWarningRepository traceTimeIntervalWarningRepo) {
+      TraceTimeIntervalWarningRepository traceTimeIntervalWarningRepo,
+      FakeCheckinsGenerator fakeCheckinsGenerator) {
     this.traceTimeIntervalWarningRepo = traceTimeIntervalWarningRepo;
+    this.fakeCheckinsGenerator = fakeCheckinsGenerator;
   }
 
   /**
    * Store the given checkin data as {@link TraceTimeIntervalWarning} entities. Returns the number
-   * of inserted entities useful for the case where there might be conflicts with the table
-   * constraints during the db save operations.
+   * of inserted entities, which is useful for the case where there might be conflicts with the
+   * table constraints during the db save operations.
    */
   @Transactional
-  public int saveCheckinData(List<CheckIn> checkins) {
+  public int saveCheckins(List<CheckIn> checkins) {
+    return saveCheckins(checkins, this::hashLocationId);
+  }
+
+  private int saveCheckins(List<CheckIn> checkins, Function<ByteString, byte[]> idHashGenerator) {
     int numberOfInsertedTraceWarnings = 0;
 
     for (CheckIn checkin : checkins) {
-      boolean traceWarningInsertedSuccessfully = traceTimeIntervalWarningRepo
-          .saveDoNothingOnConflict(checkin.getLocationId().toByteArray(),
-              checkin.getStartIntervalNumber(),
-              checkin.getEndIntervalNumber() - checkin.getStartIntervalNumber(),
-              checkin.getTransmissionRiskLevel(),
-              CheckinsDateSpecification.HOUR_SINCE_EPOCH_DERIVATION
-                  .apply(Instant.now().getEpochSecond()));
-
-      if (traceWarningInsertedSuccessfully) {
-        numberOfInsertedTraceWarnings++;
+      byte[] hashId = idHashGenerator.apply(checkin.getLocationId());
+      if (hashId != NO_HASH) {
+        boolean traceWarningInsertedSuccessfully = traceTimeIntervalWarningRepo
+            .saveDoNothingOnConflict(hashId,
+                checkin.getStartIntervalNumber(),
+                checkin.getEndIntervalNumber() - checkin.getStartIntervalNumber(),
+                checkin.getTransmissionRiskLevel(),
+                CheckinsDateSpecification.HOUR_SINCE_EPOCH_DERIVATION
+                    .apply(Instant.now().getEpochSecond()));
```

Comment:
Can lead to time zone inconsistencies during testing. Prefer to use [TimeUtils](https://github.com/corona-warn-app/cwa-server/blob/f06b7d2b0091f9fb9f6342238e31854597798fa2/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/structure/util/TimeUtils.java#L23) (which should probably be moved to "common" I guess...).